### PR TITLE
[Merged by Bors] - chore(Data/Matroid): refactoring ext lemmas

### PR DESCRIPTION
**Pull request by madvorak**
---

Please doublecheck whether I used `@[ext]` correctly.

### CODE REVIEW

**Comment by apnelson1** [comment]
LGTM

**Comment by apnelson1** [comment]
LGTM

File: Mathlib/Data/Matroid/Basic.lean [hunk 1]
```diff
@@ -665,19 +667,19 @@ theorem Base.exists_insert_of_ssubset (hB : M.Base B) (hIB : I ⊂ B) (hB' : M.B
 (hB.indep.subset hIB.subset).exists_insert_of_not_base
     (fun hI ↦ hIB.ne (hI.eq_of_subset_base hB hIB.subset)) hB'
 
-theorem eq_of_indep_iff_indep_forall {M₁ M₂ : Matroid α} (hE : M₁.E = M₂.E)
+@[ext] theorem ext_indep {M₁ M₂ : Matroid α} (hE : M₁.E = M₂.E)
     (h : ∀ I, I ⊆ M₁.E → (M₁.Indep I ↔ M₂.Indep I)) : M₁ = M₂ :=
```
Comment (apnelson1) [inline Mathlib/Data/Matroid/Basic.lean:671]: Perhaps `∀ ⦃I⦄, I ⊆ M.E → _` instead (and also in the `iff` version)? To match with `ext_base`.

File: Mathlib/Data/Matroid/Basic.lean [hunk 2]
```diff
@@ -459,7 +461,7 @@ theorem Base.diff_infinite_comm (hB₁ : M.Base B₁) (hB₂ : M.Base B₂) :
     (B₁ \ B₂).Infinite ↔ (B₂ \ B₁).Infinite :=
   infinite_iff_infinite_of_encard_eq_encard (hB₁.encard_diff_comm hB₂)
 
-theorem eq_of_base_iff_base_forall {M₁ M₂ : Matroid α} (hE : M₁.E = M₂.E)
+theorem ext_base {M₁ M₂ : Matroid α} (hE : M₁.E = M₂.E)
```
Comment (apnelson1) [inline Mathlib/Data/Matroid/Basic.lean:464]: Maybe add an `iff` version too: 

```
theorem ext_iff_base {M₁ M₂ : Matroid α} :
    M₁ = M₂ ↔ M₁.E = M₂.E ∧ ∀ ⦃B⦄, B ⊆ M₁.E → (M₁.Base B ↔ M₂.Base B) :=
  ⟨fun h ↦ by simp [h], fun ⟨hE, h⟩ ↦ ext_base hE h⟩

**Comment by madvorak** [comment]
For the sake of getting this PR reviewed, I suggest keeping it as as it is a `chore` PR and later create a `feature` PR with the additional lemmas.

**Comment by madvorak** [comment]
Ooooops I already added `ext_iff_base` last week. Should I remove it from this PR, too?

**Comment by madvorak** [comment]
> Maybe also add this one:


Since nobody was reviewing this PR anyways, I added the new lemma.

**Comment by jcommelin** [comment]
Can you please add deprecated aliases for the old names?

**Comment by madvorak** [comment]
Sorry, I always forget to do it.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**